Updates for string support and version increases on demos

amathieson opened the pull request:
+ Support for Reading and Writing OSC strings  + Bump versions on all the demos
+ Bump versions on all the demos

amathieson committed:
+ Support for Reading and Writing OSC strings
 + Bump versions on all the demos

space928 reviewed (commented):
Looks good, but we need to increment the version number before merging.

amathieson committed:
+ updated version

space928 commented on _OmsiHookExamples/BasicCLI/BasicCLI.csproj:
```diff
@@ -10,7 +10,7 @@
 
   <ItemGroup>
-    <PackageReference Include="OmsiHook" Version="2.4.4" />
+    <PackageReference Include="OmsiHook" Version="2.5.3" />
```
All of these are now going to be one version behind.

amathieson replied:
I'll bump them

space928 reviewed (commented):
Don't forget to update the version in DocFX.

amathieson committed:
Update docfx.json

amathieson committed:
+ Bump all versions

amathieson committed:
Merge branch 'String-Support' of https://github.com/space928/Omsi-Extensions into String-Support

space928 committed:
Temporarily disable automatic raw string allocation as it's broken

space928 committed:
Upgraded projects to support modern .NET:
 - String reading performance improvements
 - Windows API methods are now improtred using LibraryImport when available
 - Support for .net7 & .net8
 - Updated to C# 12
 - Various other updates
